src/pkgmt/config.py

```diff
@@ -29,72 +30,155 @@ def __len__(self) -> int:
         """
-        tag = self.cfg.get("version", {}).get("tag", None)
-        push = self.cfg.get("version", {}).get("push", None)
-        if tag is not None and not isinstance(tag, bool):
-            raise InvalidConfiguration(
-                "Type of 'tag' key in pyproject.toml is invalid. "
-                "It should be lowercase boolean : true / false"
+
+        if cli_args is None:
+            cli_args = {}
+
+        cfg_version = data.get("version", {})
+        cfg_tag = cfg_version.get("tag", None)
+        cfg_push = cfg_version.get("push", None)
+
+        tag = cli_args.get("tag") if cli_args.get("tag") is not None else cfg_tag
+        push = cli_args.get("push") if cli_args.get("push") is not None else cfg_push
+
+        keys_to_validate = ["tag", "push"]
+        values_to_validate = [tag, push]
+
+        Config._validate_boolean_key(keys_to_validate, values_to_validate)
+
+        conflicts = []
+        if cfg_tag is not None and tag != cfg_tag:
+            conflicts.append(
+                Config._generate_overriding_error("tag", tag, cfg_tag, file)
             )
-        if push is not None and not isinstance(push, bool):
-            raise InvalidConfiguration(
-                "Type of 'push' key in pyproject.toml is invalid. "
-                "It should be lowercase boolean : true / false"
+        if cfg_push is not None and push != cfg_push:
+            conflicts.append(
+                Config._generate_overriding_error("push", push, cfg_push, file)
+            )
+
+        if conflicts:
+            click.echo("\n".join(conflicts))
+
+        tag = tag if tag is not None else True
+        push = push if push is not None else True
+
+        version = {"tag": tag, "push": push}
+        data.setdefault("version", {}).update(version)
+        return data
+
+    @classmethod
+    def from_file(cls, file, **kwargs):
+        """
+        Function to generate Config object from config file.
+        Config file should contain key tool.pkgmt
+        """
+        if Path(file).exists():
+            try:
+                with open(file) as f:
+                    data = toml.load(f)["tool"]["pkgmt"]
+                    Config._validate_config(data, file)
+                    data = Config._resolve_version_configuration(
+                        data, kwargs.get("cli_args"), file
+                    )
+                    return cls(data, file)
+            except toml.decoder.TomlDecodeError as e:
+                raise InvalidConfiguration(
+                    f"Invalid {file} file: {str(e)}."
+                    "If using a boolean "
+                    "value ensure it's in lowercase, e.g., key = true"
+                ) from e
+            except KeyError as e:
+                raise InvalidConfiguration(
+                    f"Missing key : {str(e)}.\n{file} "
+                    f"should contain 'tool.pkgmt' key."
+                ) from e
+
+        else:
+            raise FileNotFoundError(
```

Comment:
pyproject.toml should be optional. does the implementation work that way?


e.g., if I have a project that doesn't have a pyproject.toml, I should still be able to use pkgmt 

also, if I have a pyproject.toml but it doesn't have [tool.pkgmt], it should still be able to use pkgmt

Reply:
But currently the pyproject.toml file is used in multiple places. Like the `github` key is used for expanding issue links in changelog. `package_name` is used for creating conda environment.

Reply:
ah, you're right. so it's mandatory.